datalad/export/tarball.py

```diff
@@ -0,0 +1,60 @@
+# emacs: -*- mode: python; py-indent-offset: 4; tab-width: 4; indent-tabs-mode: nil -*-
+# ex: set sts=4 ts=4 sw=4 noet:
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the datalad package for the
+#   copyright and license terms.
+#
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""
+
+"""
+
+__docformat__ = 'restructuredtext'
+
+import logging
+import tarfile
+import os
+from os.path import join as opj, dirname, normpath, islink, isabs
+
+lgr = logging.getLogger('datalad.export.tarball')
+
+
+# PLUGIN API
+def _datalad_export_plugin_call(dataset, output, argv=None):
+    if argv:
+        lgr.warn("tarball exporter ignores any additional options '{}'".format(
+            argv))
+
+    # could be used later on to filter files by some criterion
+    def _filter_tarinfo(ti):
+        return ti
+
+    if output is None:
+        output = "datalad_{}.tar.gz".format(dataset.id)
+    else:
+        if not output.endswith('.tar.gz.'):
+            output += '.tar.gz'
+
+    root = dataset.path
+
+    with tarfile.open(output, "w:gz") as tar:
+        for rpath in dataset.repo.get_indexed_files():
+            fpath = opj(root, rpath)
+            if islink(fpath):
```

Comment:
Maybe it would be better to check if any link points into `.git`. In direct mode we would have files under annex that are not links and cannot be resolved, so this would need to be a tested condition anyway.